### ISSUE TITLE
Add support for matrix parameters

### ISSUE THREAD

**Issue by johanhaleby** [jayway/rest-assured#417]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on November 12, 2013 15:51:03_

given().matrixParam("x", "y").when(). ..

_Original issue: http://code.google.com/p/rest-assured/issues/detail?id=263_

**Comment by Karudi** [comment]
Hello , is there any progress on this issue ? I really want to use this feature in my project , I would create a pull request but my groovy knowledge is limited . As far as I understand you can't send matrixParams directly in the url because the url will be filtered by   
 p`athWithoutQueryParams = split(tempParams, "/").inject("", pathParamFiller.curry("/", encodePath))
` , since encodePath is always set to true .
If you don't have time to solve the issue in the near future , could you suggest a workaround ? Thank you.

**Comment by johanhaleby** [comment]
Do you mean that ";" is always encoded?

**Comment by Karudi** [comment]
Yes ";" and "=". The parameter encodePath seems to always be set as true.

**Comment by johanhaleby** [comment]
Right I understand. And this is so even if you specify `given().urlEncodingEnabled(false).when(). get("http://localhost:8080/car/bmw;color=red")`?

**Comment by Karudi** [comment]
That actually works perfectly thank you .

Still would be nice to have matrixParam() support , but hey , it does the work.

**Comment by johanhaleby** [comment]
I definitely agree with that. I just need to find the time to implement it (or accept at pull request \*hint\* :))

**Comment by Ashwajeet** [comment]
Any progress on this issue

**Comment by tanner-tran-circle** [comment]
Hello from 2022. Please add this.
